### ISSUE TITLE
DHCP relay for VLANs

### ISSUE THREAD

**Issue by jbemmel** [ipspace/netlab#1343]
One can enable DHCP relay for a given interface, but not for all interfaces created for a given VLAN

Example:
```
vlans:
  red:
    dhcp.server: [ dhcp-s1, dhcp-s2 ]
```

This should result in DHCP relay being enabled for all interfaces attached to that VLAN. It also implies a check for DHCP module support for every node using that VLAN

**Comment by ipspace** [comment]
Agreed, that should work. It should be applied to the VLAN interface. Which platform(s) were you testing?

**Comment by jbemmel** [comment]
┌──────────────────────────────────────────────────────────────────────────────────┐
│ CREATING configuration files                                                     │
└──────────────────────────────────────────────────────────────────────────────────┘
[ERRORS]  Errors found in topology.yml
[ATTR]    dhcp: Invalid dhcp vlan attribute 'server' found in topology.vlans.V1234.dhcp
[HINT]    use 'netlab show attributes --module dhcp' to display valid attributes

I'm using Cumulus, but the error happens irrespective of that

**Comment by ipspace** [comment]
The problem is that the **dhcp.server** attribute is an interface, not a link attribute. After 30c8a204c74572f0564429d0ee050d29f7e13b7e you'll be able to apply it to node VLAN definition, but not to global VLAN definition.

Have to think about the implications of making DHCP attributes link attributes. It might make sense, but should come with a heavy documentation warning (imagine someone configuring **dhcp.client** and **dhcp.server** on the same VLAN)

**Comment by ipspace** [comment]
@jbemmel: I can't figure out what the best way to deal with this would be. Any suggestions are much appreciated ;)

Imagine a VLAN that has a bunch of DHCP clients and a few DHCP relays. Configuring **dhcp.client** on the VLAN sort-of makes sense (as in "we're using DHCP on this VLAN") but if we also configure **dhcp.server** on the same VLAN, what does that mean? How are we going to figure out who is a relay and who isn't?

I could implement a rule saying "if someone is a DHCP relay, they cannot also be a client" but that would still require that the **dhcp.server** is configured within node (or group) VLAN definition.
